_posts/2018-03-27-roboporo.md

```diff
@@ -0,0 +1,32 @@
+
+## Improvement and Failures
+While the fist part of the project may have taken less than a week to complete, the following took until today to hash out. Moving forward, I grew tired of having hard-coded pictures and variables, this frustration warranting the need for a configuration, or storage file instead. My solution to this involved [SnakeYAML](https://bitbucket.org/asomov/snakeyaml), a YAML utility able to parse YAML file types into java Objects, the perfect solution to my issue. Once the configuration file was set up, I expanded upon the functionality of the program, adding a list command, the ability to add and remove entries from the configuration file, and even throwing in a novelty /roll command to add non-present convenience to the servers I was on.
+
+My final problem for this project came in the way of multi-server support however. When I attempted to place this bot onto multiple servers, I learned that they all shared the same configuration file - causing something of an issue when people wanted to keep images private in their communities instead of having them spread around. My solution to this was the dynamic generation of configuration files, accomplished through the use of an excessive amount of HashMaps, and adding a link between a given Guild's unique ID and the configuration file it was responsable for. At this innovation, the bot worked well above and beyond what I could have hoped to have accomplished from the start, and I was satisfied with its progress.
+
+## Lessons and Moving forward
+This project allowed me to learn plenty about the utilization of API's in a manner which was both fun, and useful to my everyday life on Discord. Though I originally considered myself somewhat versed in Java, the completion of RoboPoro also has further cemented my knowledge in its file handling and proper implementation procedures as well. Moving forward with this project, I have a few ideas in mind, the most notable being the act of moving this bot to a stable host - potentially through OpenShift or on a dedicated server elsewhere. Additionally, expanding upon RoboPoro's functionality to include more utility commands (getting game stats, fetching images from the internet instead of configs, etc.) is also on the table, and likely to be my next step in development.
```

Comment:
`s/API's/APIs/`
`s/both fun, and/both fun and/`
`s/versed/well-versed/`
`s/further cemented my knowledge in/further improved my knowledge of/`
`s/the act of//`